azurelinuxagent/ga/exthandlers.py

```diff
@@ -241,6 +241,7 @@ class ExtHandlerState(object):
     Installed = "Installed"
     Enabled = "Enabled"
     FailedUpgrade = "FailedUpgrade"
+    DidNotRun = "DidNotRun"
```

Comment:
I do not like the idea of creating a new state for this case. It seems like an overkill for what we're trying to achieve since creating and maintaining a new state requires dealing with the filesystem which comes with its own issues. Additionally, I don't think it fits the needs completely either. 

For eg, you're setting the HandlerState for Single config extension to `DidNotRun` but not for MultiConfig extensions. This mismatch makes the code more prone to errors.

Reply:
Maybe save this state in-memory and use it when its about to be reported? For example modifying the `extHandler` object because that's common for both the extension processing and reporting and can potentially be used for this case?